atmat/lattice/survey/atgeometry.m

```diff
@@ -20,29 +20,31 @@
 %       a scalar offset value is equivalent to [0 OFFSET]
 %
 %POSDATA=ATGEOMETRY(...,'centered')
-%       The offset is set as [0 RADIUS 0]
+%       The offset is set as [0 RADIUS]
 %
 %POSDATA=ATGEOMETRY(...,'Hangle',h_angle)
 %       Set the initial trajectory angle
 %
 %
 %See also: ATGEOMETRY3
 
-[thetac,args]=getoption(varargin,'Hangle',0);
-centered=getflag(args,'centered');
-[refpts,offset]=parseargs({1:length(ring)+1,[0 0]},args);
+[centered,args]=getflag(varargin,'centered');
+[theta0,args]=getoption(args,'Hangle',0);
+[refpts,offset]=getargs(args,1:length(ring)+1,[0 0]);
 xc=0;
 yc=0;
+thetac=theta0;
 if isscalar(offset)
     yc=offset;
 else
     xc=offset(1);
     yc=offset(2);
 end
 [xx,yy,txy,lg,tr]=cellfun(@incr,[{struct()};ring]);    % Add a dummy element to get the origin
-radius=-(yc+xc/tan(thetac));
+radius=-(yc+xc/tan(thetac-theta0));
 if centered
-    yy=yy+radius;
+    xx=xx-radius*sin(theta0);
```

Comment:
maybe i'm missing something. But why not do
```
xcenter = (max(xx) + min(xx) )/2;
ycenter = (max(yy) + min(yy) )/2;
xx=xx-xcenter;
yy=yy-ycenter;
```

Reply:
@lcarver would work only for full SR.